MetaMask/metamask-extension
# UX: Reduce setMouseUserState calls to prevent re-renders

## **Description**
While debugging another issue, I discovered that every time I clicked anywhere on the screen, there would be an app re-render, which felt unnecessary and as though it could cause unwanted performance hits.  I believe I've tracked it down to `setMouseUserState` in `routes.component.js`, which would be dispatched any time a user clicked anywhere in the UI.  The original PR cited https://github.com/MetaMask/metamask-extension/pull/6860 .  This methodology is very outdated, as it sets the following styles via a classname upon click:

```
# ui/css/base-styles.scss
.mouse-user-styles {
  button:focus,
  input:focus,
  textarea:focus,
  .unit-input__input,
  .currency-display__input {
    outline: none;
  }
}
```

I also believe the new Design System probably covers all of this, so we may even be able to remove the listener as a whole.  In any event, this PR makes it such that the dispatch only happens the first time the user uses their mouse, which should be a big perf win.

## **Manual testing steps**

1.  Look at the page code not have `mouse-user-styles`
2. Click anywhere in the document
3. Open devtools, set a breakpoint at `ui/pages/routes/routes.component.js`
4. Click again, don't see breakpoint hit.

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
